app/tests/hanging_protocols_tests/test_forms.py

```diff
@@ -666,3 +669,65 @@ def test_generate_view_content_example(
     )
 
     assert view_content_example_json == expected_example_json
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "hanging_protocol_json",
```

Comment:
The test function creates a civ_list with 4 component interfaces that can be set in a view port (images+isolated). The parameterized test covers the following cases:
- No viewports in the hanging protocol json
- The hanging protocol viewports are less than the number of CIs (images+isolated)
- The hanging protocol viewports are equal to the number of CIs (images+isolated) 
- The hanging protocol viewports are larger than the number of CIs (images+isolated)

Reply:
outdated